src/util/patientUtils.js

```diff
@@ -9,11 +9,14 @@ const { mapResourcesToCollectionBundle, mapArrayToSearchSetBundle } = require('.
 /**
  * Wrapper function to get patient data and data
  * requirements for given patient id and map the resources to a collection bundle.
- * @param {string} patientId patient ID of interest
+ * @param {string} patientReference reference to patient of interest (either of the form {PatientId} or Patient/{PatientId})
  * @param {Array} dataRequirements data requirements array obtained from fqm execution
  * @returns {Object} patient bundle as a collection bundle
  */
-async function getPatientDataCollectionBundle(patientId, dataRequirements) {
+async function getPatientDataCollectionBundle(patientReference, dataRequirements) {
+  // PatientReference can be of the form {PatientId} or Patient/{PatientId} this extracts just the PatientId in both cases
+  const splitRef = patientReference.split('/');
+  const patientId = splitRef[splitRef.length - 1];
```

Comment:
Do we want a check here that the first element in the array is 'Patient' or is that overkill?

Reply:
I think this is okay for now. If we start passing other resource type references around at some point, it might then be worth throwing a special error for the wrong kind of resource. But since this is the only one of this style for now, I tend to think that the user will understand what's wrong well enough...